packages/@aws-cdk/aws-lambda-event-sources/lib/dynamodb.ts

```diff
@@ -1,5 +1,6 @@
 import dynamodb = require('@aws-cdk/aws-dynamodb');
 import lambda = require('@aws-cdk/aws-lambda');
+import { Duration } from "@aws-cdk/core";
 
 export interface DynamoEventSourceProps {
```

Comment:
`DynamoEventSourceProps` and `KinesisEventSourceProps`  contain the exact same set of attributes, with the same defaults.

Perhaps we should define a base interface `StreamingEventSourceProps` with these attributes that `DynamoEventSourceProps` and `KinesisEventSourceProps` extend (both of which will be empty to begin with).

What do you think?

Reply:
Completely agree, I'm on it 👍

Reply:
I ended up going a little harder on the refactor and created a `StreamEventSource` class. I'll add tests for it if you're on board